### DISCUSSION TITLE
Expose received Websocket Handshake Headers

### DISCUSSION THREAD

**Discussion by Nikola-Milovic** [godotengine/godot-proposals#8708] [General Discussions]
Current implementation of [`WebsocketPeer.HandshakeHeaders`](https://docs.godotengine.org/en/stable/classes/class_websocketpeer.html#class-websocketpeer-property-handshake-headers) says the following

```
string[] WebSocketPeer.HandshakeHeaders { get; set; }

The extra HTTP headers to be sent during the WebSocket handshake\.

Note: Not supported in Web exports due to browsers' restrictions\.
```

Meaning this is only supported to *set* the headers when establishing a connection, but since the you can also use this peer on the server side, it would be useful to *receive* the headers as well. For example if I am implementing a WS server in Godot I would like to have a way to get the auth headers that occured during the handshake.

**Comment by Az-qianchen** [comment]
If we can receive it, maybe we can use it as an authorization?

**Reply by Nikola-Milovic** [reply]
Yeah that lead me to this proposal. With Godot WS you cannot do handshake authorization since you have no access to the tokens :/
